content_store_controller.go

```diff
@@ -75,23 +91,52 @@ func (cs *ContentStoreController) registerWithURLArbiter(path, publishingApp str
 	return true
 }
 
-func (cs *ContentStoreController) GetContentStoreRequest(w http.ResponseWriter, r *http.Request) {
-	cs.doContentStoreRequest("GET", r.URL.Path, nil, w)
-}
-
-func (cs *ContentStoreController) DeleteContentStoreRequest(w http.ResponseWriter, r *http.Request) {
-	cs.doContentStoreRequest("DELETE", r.URL.Path, nil, w)
-}
-
 // data will be nil for requests without bodies
-func (cs *ContentStoreController) doContentStoreRequest(httpMethod string, path string, data []byte, w http.ResponseWriter) {
-	resp, err := cs.contentStore.DoRequest(httpMethod, path, data)
+func (controller *ContentStoreController) doContentStoreRequest(httpMethod string, path string, data []byte, w http.ResponseWriter) {
+	resp, err := controller.liveContentStore.DoRequest(httpMethod, path, data)
+	defer resp.Body.Close()
+
 	if err != nil {
 		renderer.JSON(w, http.StatusInternalServerError, err)
 	}
-	defer resp.Body.Close()
 
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(resp.StatusCode)
 	io.Copy(w, resp.Body)
 }
+
+func (controller *ContentStoreController) doDraftContentStoreRequest(httpMethod string, path string, data []byte, w http.ResponseWriter) {
+	resp, err := controller.draftContentStore.DoRequest(httpMethod, path, data)
+	defer resp.Body.Close()
+
+	if w != nil {
+		if err != nil {
+			renderer.JSON(w, http.StatusInternalServerError, err)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(resp.StatusCode)
+		io.Copy(w, resp.Body)
+	}
+}
```

Comment:
This method is very similar to `doContentStoreRequest`.  The only difference being which content-store it writes to, and handling `w` being nil.  It's probably worth combining these 2 methods into a single one that takes the content store client instance as a parameter.  I don't think it would be a problem to have the `w` nil handling for both cases.